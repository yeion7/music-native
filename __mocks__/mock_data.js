export const status = {
  showPlayer: false,
  expand: false,
  isLoading: false,
  isPlaying: false,
  position: null,
  duration: null
};

export const song = {
  id: "0dMDz3n9qLfRXBNlRhK5U7",
  name: "Joe Tex, These Taming Blues",
  duration_ms: 330813,
  preview_url: "https://p.scdn.co/mp3-preview/45d2ad042bf08d7951076097a1f7f15677ab7a89?cid=null",
  artists: [
    {
      external_urls: {
        spotify: "https://open.spotify.com/artist/57kIMCLPgkzQlXjblX7XXP"
      },
      href: "https://api.spotify.com/v1/artists/57kIMCLPgkzQlXjblX7XXP",
      id: "57kIMCLPgkzQlXjblX7XXP",
      name: "Phosphorescent",
      type: "artist",
      uri: "spotify:artist:57kIMCLPgkzQlXjblX7XXP"
    }
  ],
  album: {
    album_type: "album",
    artists: [
      {
        external_urls: {
          spotify: "https://open.spotify.com/artist/57kIMCLPgkzQlXjblX7XXP"
        },
        href: "https://api.spotify.com/v1/artists/57kIMCLPgkzQlXjblX7XXP",
        id: "57kIMCLPgkzQlXjblX7XXP",
        name: "Phosphorescent",
        type: "artist",
        uri: "spotify:artist:57kIMCLPgkzQlXjblX7XXP"
      }
    ],
    external_urls: {
      spotify: "https://open.spotify.com/album/0TticAuqNj26EhgT2Em3qD"
    },
    href: "https://api.spotify.com/v1/albums/0TticAuqNj26EhgT2Em3qD",
    id: "0TticAuqNj26EhgT2Em3qD",
    images: [
      {
        height: 640,
        url: "https://i.scdn.co/image/ccd62a1aa5c66d581622a1be2c86cfd0299a3b45",
        width: 640
      },
      {
        height: 300,
        url: "https://i.scdn.co/image/cc9f4c667dc6fa65839c121262a7e7ba5dd49547",
        width: 300
      },
      {
        height: 64,
        url: "https://i.scdn.co/image/1ea5645d08a452f50c9720aad99d4ea6e8cdd8b6",
        width: 64
      }
    ],
    name: "Aw Come Aw Wry",
    type: "album",
    uri: "spotify:album:0TticAuqNj26EhgT2Em3qD"
  },
  track_number: 3
};

export const playlist = [
  {
    id: "0dMDz3n9qLfRXBNlRhK5U7",
    name: "Joe Tex, These Taming Blues",
    duration_ms: 330813,
    preview_url: "https://p.scdn.co/mp3-preview/45d2ad042bf08d7951076097a1f7f15677ab7a89?cid=null",
    artists: [
      {
        external_urls: {
          spotify: "https://open.spotify.com/artist/57kIMCLPgkzQlXjblX7XXP"
        },
        href: "https://api.spotify.com/v1/artists/57kIMCLPgkzQlXjblX7XXP",
        id: "57kIMCLPgkzQlXjblX7XXP",
        name: "Phosphorescent",
        type: "artist",
        uri: "spotify:artist:57kIMCLPgkzQlXjblX7XXP"
      }
    ],
    album: {
      album_type: "album",
      artists: [
        {
          external_urls: {
            spotify: "https://open.spotify.com/artist/57kIMCLPgkzQlXjblX7XXP"
          },
          href: "https://api.spotify.com/v1/artists/57kIMCLPgkzQlXjblX7XXP",
          id: "57kIMCLPgkzQlXjblX7XXP",
          name: "Phosphorescent",
          type: "artist",
          uri: "spotify:artist:57kIMCLPgkzQlXjblX7XXP"
        }
      ],
      external_urls: {
        spotify: "https://open.spotify.com/album/0TticAuqNj26EhgT2Em3qD"
      },
      href: "https://api.spotify.com/v1/albums/0TticAuqNj26EhgT2Em3qD",
      id: "0TticAuqNj26EhgT2Em3qD",
      images: [
        {
          height: 640,
          url: "https://i.scdn.co/image/ccd62a1aa5c66d581622a1be2c86cfd0299a3b45",
          width: 640
        },
        {
          height: 300,
          url: "https://i.scdn.co/image/cc9f4c667dc6fa65839c121262a7e7ba5dd49547",
          width: 300
        },
        {
          height: 64,
          url: "https://i.scdn.co/image/1ea5645d08a452f50c9720aad99d4ea6e8cdd8b6",
          width: 64
        }
      ],
      name: "Aw Come Aw Wry",
      type: "album",
      uri: "spotify:album:0TticAuqNj26EhgT2Em3qD"
    },
    track_number: 3
  }
];

export const albums = [
  {
    id: "1bypUQ8qclz4N9ocZgAmPj",
    name: "Aw Man Oh Dear",
    image: "https://i.scdn.co/image/172a34d390848ba1615374fa4f77c304de01a919",
    artists: {
      external_urls: {
        spotify: "https://open.spotify.com/artist/77PtLt3o5UjeAT38mVlGaQ"
      },
      href: "https://api.spotify.com/v1/artists/77PtLt3o5UjeAT38mVlGaQ",
      id: "77PtLt3o5UjeAT38mVlGaQ",
      name: "Aw Man Oh Dear",
      type: "artist",
      uri: "spotify:artist:77PtLt3o5UjeAT38mVlGaQ"
    },
    url_songs: "https://api.spotify.com/v1/albums/1bypUQ8qclz4N9ocZgAmPj",
    type: "album"
  }
];
