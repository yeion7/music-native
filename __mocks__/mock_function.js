export const Mock = jest.fn();
