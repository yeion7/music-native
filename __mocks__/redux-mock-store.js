import configureMockStore from "redux-mock-store";

const middlewares = [];
const mockStore = configureMockStore(middlewares);

export default mockStore;
