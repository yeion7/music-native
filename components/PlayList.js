/* @flow weak */

import React from "react";
import { View, Dimensions } from "react-native";
import { Text, List, ListItem, Left, Right } from "native-base";
import styled from "styled-components/native";

const { width } = Dimensions.get("window");

import { formattedTime } from "../lib/time";
const PlayList = ({ playList, onPressSong }) => (
  <Wrapper>
    <List
      dataArray={playList}
      renderRow={song => (
        <ListItem key={song.id} onPress={() => onPressSong(song)}>
          <Left>
            <Text>{song.name}</Text>
          </Left>
          <Right>
            <Text>{formattedTime(song.duration_ms)}</Text>
          </Right>
        </ListItem>
      )}
    />
  </Wrapper>
);

const Wrapper = styled.View`
  width: ${width}
`;
export default PlayList;
