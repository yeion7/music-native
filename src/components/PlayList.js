/* @flow weak */

import React from "react";
import { View, Dimensions } from "react-native";
import { List } from "native-base";

import styled from "styled-components/native";
import ItemContainer from "../containers/ItemContainer";

const { width, height } = Dimensions.get("window");

const RenderItem = song => <ItemContainer {...song} key={song.id} />;

const PlayList = ({ playList }) => (
  <Wrapper>
    <List dataArray={playList} renderRow={RenderItem} />
  </Wrapper>
);

const Wrapper = styled.View`
  flex:1;
  width: ${width};
  height: ${height};
`;
export default PlayList;
