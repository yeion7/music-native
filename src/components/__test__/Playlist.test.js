import React from "react";
import "react-native";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";

import { Mock } from "../../../__mocks__/mock_function";
import { playlist } from "../../../__mocks__/mock_data";
import mockStore from "../../../__mocks__/redux-mock-store";

import PlayList from "../PlayList";

const store = mockStore({});

describe("renders test ", () => {
  it("render PlayList component", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PlayList playList={playlist} onPressSong={Mock} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
