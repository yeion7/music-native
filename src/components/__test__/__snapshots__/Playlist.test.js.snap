// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders test  render PlayList component 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 667,
        "width": 375,
      },
      undefined,
    ]
  }
>
  <ScrollView
    dataArray={
      Array [
        Object {
          "album": Object {
            "album_type": "album",
            "artists": Array [
              Object {
                "external_urls": Object {
                  "spotify": "https://open.spotify.com/artist/57kIMCLPgkzQlXjblX7XXP",
                },
                "href": "https://api.spotify.com/v1/artists/57kIMCLPgkzQlXjblX7XXP",
                "id": "57kIMCLPgkzQlXjblX7XXP",
                "name": "Phosphorescent",
                "type": "artist",
                "uri": "spotify:artist:57kIMCLPgkzQlXjblX7XXP",
              },
            ],
            "external_urls": Object {
              "spotify": "https://open.spotify.com/album/0TticAuqNj26EhgT2Em3qD",
            },
            "href": "https://api.spotify.com/v1/albums/0TticAuqNj26EhgT2Em3qD",
            "id": "0TticAuqNj26EhgT2Em3qD",
            "images": Array [
              Object {
                "height": 640,
                "url": "https://i.scdn.co/image/ccd62a1aa5c66d581622a1be2c86cfd0299a3b45",
                "width": 640,
              },
              Object {
                "height": 300,
                "url": "https://i.scdn.co/image/cc9f4c667dc6fa65839c121262a7e7ba5dd49547",
                "width": 300,
              },
              Object {
                "height": 64,
                "url": "https://i.scdn.co/image/1ea5645d08a452f50c9720aad99d4ea6e8cdd8b6",
                "width": 64,
              },
            ],
            "name": "Aw Come Aw Wry",
            "type": "album",
            "uri": "spotify:album:0TticAuqNj26EhgT2Em3qD",
          },
          "artists": Array [
            Object {
              "external_urls": Object {
                "spotify": "https://open.spotify.com/artist/57kIMCLPgkzQlXjblX7XXP",
              },
              "href": "https://api.spotify.com/v1/artists/57kIMCLPgkzQlXjblX7XXP",
              "id": "57kIMCLPgkzQlXjblX7XXP",
              "name": "Phosphorescent",
              "type": "artist",
              "uri": "spotify:artist:57kIMCLPgkzQlXjblX7XXP",
            },
          ],
          "duration_ms": 330813,
          "id": "0dMDz3n9qLfRXBNlRhK5U7",
          "name": "Joe Tex, These Taming Blues",
          "preview_url": "https://p.scdn.co/mp3-preview/45d2ad042bf08d7951076097a1f7f15677ab7a89?cid=null",
          "track_number": 3,
        },
      ]
    }
    dataSource={
      ListViewDataSource {
        "items": 1,
      }
    }
    enableEmptySections={true}
    initialListSize={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    removeClippedSubviews={true}
    renderRow={[Function]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []}
    stickySectionHeadersEnabled={true}
    style={Object {}}
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flexDirection": "row",
            "marginLeft": 15,
            "paddingBottom": 13,
            "paddingRight": 15,
            "paddingTop": 13,
          }
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
                "marginLeft": 10,
              }
            }
            virtual={undefined}
          >
            Joe Tex, These Taming Blues
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 0.28,
              "paddingBottom": null,
              "paddingLeft": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": null,
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            virtual={undefined}
          >
            5:30
          </Text>
        </View>
      </View>
    </View>
  </ScrollView>
</View>
`;
